lesson18/src/js/main.js

```diff
@@ -0,0 +1,273 @@
+  const nextButton = document.getElementById("js-button_next");
+
+  prevButton.disabled = Number(index) === firstIndex;
+  nextButton.disabled = Number(index) === lastIndex;
+}
+
+function renderSlidContents(slideImage){
+  renderSlideImage(slideImage);
+  renderPagination(slideImage.length);
+  renderPaginationBullet(slideImage.length);
+  renderSlideButton();
+}
+
+function addSlideAction(){
+  addChangeButtonEvent();
+  clickPaginationBullet();
+  autoPlayslide();
+}
+
+function accessSlideImage() {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(
+        fetchSlideImages(
+          // Comment out and leave the API URL of 503 and 0 images as the URL for checking error handling.
+          //image = 0
+          //"https://api.json-generator.com/templates/IZjWl012CAMD/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl"
+          //503 error
+          //"https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl&status=503"
+          "https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl"
+        )
+      );
+    }, 3000);
+  });
+}
+
+async function fetchSlideImages(URL) {
+  const response = await fetch(URL);
+  if (response.ok) {
+    const json = await response.json();
+    return json;
+  } else {
+    console.error(`${response.status}:${response.statusText}`);
+    displayErrorMassage("Internet Server Error");
+  }
+}
+
+function addChangeButtonEvent() {
+  const arrowButtons = document.querySelector(".slider__button");
+
+  arrowButtons.addEventListener("click", (event) => {
+    const activeImage = document.querySelector(".is-active");
+    let currentIndex = Number(activeImage.dataset.slideIndex);
+    currentIndex =
+      event.target.id === "js-button_next" ? ++currentIndex : --currentIndex;
+    changeActivePagination(currentIndex);
+    changeActiveSlider(currentIndex);
+    resetAutoPlaySlide()
+  });
+}
+
+function changeActiveSlider(slideTarget) {
+  toggleButtonDisabled(slideTarget);
+  changeActiveImage(slideTarget);
+  changeActivePaginationBullet(slideTarget);
+}
+
+let autoPlayID;
+function autoPlayslide() {
+  autoPlayID = setInterval(() => {
+    const slideImages = document.querySelectorAll('.slider__item').length;
+    const activeImage = document.querySelector('.is-active');
+    let currentIndex = Number(activeImage.dataset.slideIndex);
+    currentIndex = currentIndex === slideImages ? 1 : currentIndex + 1;
+    changeActiveSlider(currentIndex);
+    changeActivePagination(currentIndex);
+  },3000)
+}
+
+function resetAutoPlaySlide(){
+  clearInterval(autoPlayID);
+  autoPlayslide();
+}
+
+async function init() {
+  showLoadingImage();
+  renderSlideArea();
+  let slideImages;
+  try{
+     const json = await accessSlideImage();
+     slideImages = json.slide;
+  }catch{
+    console.error('There was no image');
```

Comment:
IMO
I think the error you want to catch here is when data could not be retrieved due to server error, etc. I think the processing for the case where no image was found has already been implemented with an if statement.

Reply:
Thank you for your review!
If the error value is fixed, you will not know which error occurred, so
Changed to take an argument.

[9a2a3ce](https://github.com/nyattsu72/hanson-frontend-cource/pull/26/commits/9a2a3cea7682a038758a54c5cb45031f3293db03)